x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings/es_fields_mappings_helpers.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Fields } from './types';
+
+export const normalizeValue = <T>(field: unknown[] | unknown): T => {
+  return (Array.isArray(field) && field.length === 1 ? field[0] : field) as T;
```

Comment:
I don't think we should do it like this, we need an allowlist. otherwise what will happen is that things which are for instance known multi-value fields will sometimes be a plain value and sometimes an array

Reply:
I wanted to make it more generic and support arrays as well but I checked and so far we don't have any array values in our mappings - So to keep it simple I will revert this change and once we want to map array values we can implement it specific for multi-value fields.

Reply:
yes we do, `process.args` and `child.id`

Reply:
They are not part of any of the mappers we have, do you know exactly where we have them (and which query and which APM type so we can update accordingly)?